src/OmniSharp.Roslyn.CSharp/Services/Signatures/SignatureHelpService.cs

```diff
@@ -96,27 +97,43 @@ public SignatureHelpService(OmniSharpWorkspace workspace)
                 var invocation = node as InvocationExpressionSyntax;
                 if (invocation != null && invocation.ArgumentList.Span.Contains(position))
                 {
+                    var i = await document.GetSemanticModelAsync();
                     return new InvocationContext()
                     {
                         SemanticModel = await document.GetSemanticModelAsync(),
                         Position = position,
                         Receiver = invocation.Expression,
-                        ArgumentList = invocation.ArgumentList
+                        ArgumentTypes = invocation.ArgumentList.Arguments.Select(argument => i.GetTypeInfo(argument.Expression)),
+                        Separators = invocation.ArgumentList.Arguments.GetSeparators()
                     };
                 }
-
                 var objectCreation = node as ObjectCreationExpressionSyntax;
                 if (objectCreation != null && objectCreation.ArgumentList.Span.Contains(position))
                 {
+                    var i = await document.GetSemanticModelAsync();
                     return new InvocationContext()
                     {
-                        SemanticModel = await document.GetSemanticModelAsync(),
+                        SemanticModel = i,
                         Position = position,
                         Receiver = objectCreation,
-                        ArgumentList = objectCreation.ArgumentList
+                        ArgumentTypes = objectCreation.ArgumentList.Arguments.Select(argument => i.GetTypeInfo(argument.Expression)),
```

Comment:
Same as above

Reply:
@akshita31 we can extract this logic into an extension method or a private function.

Reply:
I'd consider adding two constructor overloads to InvocationContext which will deal with this initialization. If we want to keep InvocationContext a POCO [Plain Old C# Object] then we can put this logic into a factory method instead. I don't have a strong preference.